src/linalg/basic/matrix.rs

```diff
@@ -49,27 +51,75 @@ pub struct DenseMatrixMutView<'a, T: Debug + Display + Copy + Sized> {
     column_major: bool,
 }
 
+// functional utility functions used across types
```

Comment:
@EdmundsEcho
Please follow Rust code conventions, every functions shall be documented using triple slashes `///`. I know this is boring and nitpicking but we have to stick to conventions. Also the content of the documentation shall be explicative of what the function does; "functional utilities" is very generic.